risc0/zkvm/src/exec/mod.rs

```diff
@@ -129,6 +130,11 @@ impl Write for Journal {
 
 impl<'a> Executor<'a> {
     /// Construct a new [Executor] from a [MemoryImage] and entry point.
+    /// Before a guest program is proven, the [Executor] is responsible for
+    /// deciding where a zkVM program should be split into [Segment]s and what
+    /// work will be done in each segment. This is the execution phase:
+    /// the guest program is executed to determine how its proof should be
+    /// divided into subparts.
```

Comment:
```suggestion
    /// Construct a new [Executor] from a [MemoryImage] and entry point.
    ///
    /// Before a guest program is proven, the [Executor] is responsible for
    /// deciding where a zkVM program should be split into [Segment]s and what
    /// work will be done in each segment. This is the execution phase:
    /// the guest program is executed to determine how its proof should be
    /// divided into subparts.
```

Reply:
I think how the executor works should be described on the `Executor` and not the constructor. I also think execution is more than merely deciding where splits occur. It also consumes all the inputs to generate a trace that the prover can use.